client/state/data-layer/wpcom/sites/plan-transfer/index.js

```diff
@@ -0,0 +1,70 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { dispatchRequestEx } from 'state/data-layer/wpcom-http/utils';
+import { errorNotice, successNotice } from 'state/notices/actions';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { SITE_PLAN_OWNERSHIP_TRANSFER } from 'state/action-types';
+
+const noticeOptions = siteId => ( {
+	duration: 8000,
+	id: `sites-plan-transfer-notice-${ siteId }`,
+} );
+
+/**
+ * Dispatches a request to transfer a site's plan to another user.
+ *
+ * @param   {Object} action Redux action
+ * @returns {Object} Dispatched http action
+ */
+export const requestPlanOwnershipTransfer = action =>
+	http(
+		{
+			apiVersion: '1',
+			method: 'POST',
+			path: '/sites/' + action.siteId + '/plan-transfer',
+			query: {
+				new_user_id: action.newUserId,
+			},
+		},
+		action
+	);
+
+/**
+ * Dispatches a success notice when the request succeeded.
+ *
+ * @param   {Object} action Redux action
+ * @returns {Object} Success notice action
+ */
+export const handleTransferSuccess = ( { siteId } ) =>
+	successNotice(
+		translate( 'Plan purchaser has been changed successfully.' ),
```

Comment:
This notice could be improved, but maybe that can be changed when the UI lands and we might now the other user name or some site identifier to put in the message?

Reply:
I've intentionally tried to keep this one generic - we can iterate on it in the future.